test/app_test_case.py

```diff
@@ -65,14 +68,19 @@ def lambda_name(cls) -> str:
         """
         raise NotImplementedError
 
-    @property
-    def base_url(self):
+    def base_url(self, endpoint_path=None, **params) -> str:
```

Comment:
Depending on the parameters, this doesn't really return a "base" URL anymore. I would change this back to a property, have it return a furl object and then further manipulate the furl object where needed. In fact, it seems that with this change you end up having to touch a lot of lines but that all those call sites continue to compose URLs improperly by concatenation. Please undo all of those changes and only make minimal changes to the manifest test.